File: wiki-backend/src/main/java/com/internal/team/wiki/doc/dto/request/DocCreateRequest.java
package com.internal.team.wiki.doc.dto.request;

import com.internal.team.wiki.doc.domain.DocEntity;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;

@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class DocCreateRequest {

    @NotBlank
    private String writerName;

    @NotBlank
    private String title;

    @NotBlank
    private String contents;

    @Builder
    public DocCreateRequest(final String writerName, final String title, final String contents) {
        this.writerName = writerName;
        this.title = title;
        this.contents = contents;
    }

    public DocEntity toEntity(final DocCreateRequest request) {
        return DocEntity.builder()
                .writerName(request.getWriterName())
                .title(request.getTitle())
                .contents(request.getContents())
                .build();

    }
}

File: wiki-backend/src/main/java/com/internal/team/wiki/doc/dto/DocCreateRequest.kt
package com.internal.team.wiki.doc.dto

import com.internal.team.wiki.doc.domain.DocEntity
import com.internal.team.wiki.user.domain.UserEntity
import javax.validation.constraints.NotBlank

data class DocCreateRequest(

    @field:NotBlank
    val writerName: String,

    @field:NotBlank
    val title: String,

    @field:NotBlank
    val contents: String
) {
    fun toEntity(user: UserEntity, request: DocCreateRequest): DocEntity {
        return DocEntity(
            user = user,
            writerName = user.nickname.value,
            title = request.title,
            contents = request.contents
        )
    }
}
